### DISCUSSION TITLE
How to change some parameters in EMQX plugin configuration according to environment variables

### DISCUSSION THREAD

**Discussion by hokaso** [emqx/emqx#8335] [Q&A] [answered]
### Environment Information

- EMQX version: 4.3.10
- Operating system and version: Linux debian9

### Problem Description

The first is that the business needs to enable the webhook plug-in, and then our devops involves multiple environments. The web.hook.url configured in each environment is different. When the service is deployed in different environments, it needs to be read according to the environment variables read. To change some parameters in the EMQX plugin configuration, it seems that this step is not listed in the official documents, so come and ask the big guys~

### 环境信息

- EMQX 版本：4.3.10
- 操作系统及版本：Linux debian9

### 问题描述

首先是业务需要启用webhook插件，然后我们的devops中涉及多个环境，每个环境配置的web.hook.url都不一样，当服务部署到不同的环境中时，需要根据读取到的环境变量来改变EMQX插件配置中的某些参数，就是这一步似乎官方文档上没有列出，所以过来询问一下大佬们~

**Comment by lafirest** [comment, marked as answer]
https://www.emqx.io/docs/en/v4.4/configuration/environment-variable.html
@hokaso Hi, maybe this is what you want
